src/components/ExpensiTextInput/index.ios.js

```diff
@@ -0,0 +1,126 @@
+import React, {Component} from 'react';
+import {
+    Animated, TextInput,
+} from 'react-native';
+import ExpensiTextInputWrapper from './ExpensiTextInputWrapper';
+import ExpensiTextInputLabel from './ExpensiTextInputLabel';
+import expensiTextInputPropTypes from './ExpensiTextInputPropTypes';
+import expensiTextInputDefaultProps from './ExpensiTextInputDefaultProps';
+import styles from '../../styles/styles';
+import themeColors from '../../styles/themes/default';
+
+const ACTIVE_LABEL_TRANSLATE_Y = -10;
+const ACTIVE_LABEL_TRANSLATE_X = (translateX = -22) => translateX;
+const ACTIVE_LABEL_SCALE = 0.8668;
+
+const INACTIVE_LABEL_TRANSLATE_Y = 0;
+const INACTIVE_LABEL_TRANSLATE_X = 0;
+const INACTIVE_LABEL_SCALE = 1;
+
+class ExpensiTextInput extends Component {
+    constructor(props) {
+        super(props);
+
+        const hasValue = props.value.length > 0;
+
+        this.state = {
+            isFocused: false,
+            labelTranslateY: new Animated.Value(hasValue ? ACTIVE_LABEL_TRANSLATE_Y : INACTIVE_LABEL_TRANSLATE_Y),
+            labelTranslateX: new Animated.Value(hasValue
+                ? ACTIVE_LABEL_TRANSLATE_X(props.translateX) : INACTIVE_LABEL_TRANSLATE_X),
+            labelScale: new Animated.Value(hasValue ? ACTIVE_LABEL_SCALE : INACTIVE_LABEL_SCALE),
+        };
+
+        this.input = null;
+    }
+
+    animateLabel = (translateY, translateX, scale) => {
+        Animated.parallel([
+            Animated.spring(this.state.labelTranslateY, {
+                toValue: translateY,
+                duration: 80,
+                useNativeDriver: true,
+            }),
+            Animated.spring(this.state.labelTranslateX, {
+                toValue: translateX,
+                duration: 80,
+                useNativeDriver: true,
+            }),
+            Animated.spring(this.state.labelScale, {
+                toValue: scale,
+                duration: 80,
+                useNativeDriver: true,
+            }),
+        ]).start();
+    }
+
+    onFocus = () => {
+        if (this.props.onFocusExtra) { this.props.onFocusExtra(); }
+        this.setState({isFocused: true});
+        if (this.props.value.length === 0) {
+            this.animateLabel(
+                ACTIVE_LABEL_TRANSLATE_Y,
+                ACTIVE_LABEL_TRANSLATE_X(this.props.translateX),
+                ACTIVE_LABEL_SCALE,
+            );
+        }
+    }
+
+    onBlur = () => {
+        if (this.props.onBlurExtra) { this.props.onBlurExtra(); }
+        this.setState({isFocused: false});
+        if (this.props.value.length === 0) {
+            this.animateLabel(INACTIVE_LABEL_TRANSLATE_Y, INACTIVE_LABEL_TRANSLATE_X, INACTIVE_LABEL_SCALE);
+        }
+    }
+
+    focus = () => this.input.focus()
```

Comment:
I don't think we need this function, but if you _do_ include it make it a normal function instead of an arrow function.

Reply:
Removed